desktop/core/src/desktop/js/reactComponents/Toolbar/Toolbar.test.tsx

```diff
@@ -0,0 +1,206 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+
+import ImportIcon from '../../components/icons/ImportIcon';
+
+import Toolbar, { ToolbarButton, ToolbarDivider } from './Toolbar';
+
+describe('Toolbar', () => {
+  test('renders any content', () => {
+    render(<Toolbar content={() => <div>custom content</div>} />);
+    expect(screen.getByText('custom content')).toBeVisible();
+  });
+  test('has default test-id "hue-toolbar"', () => {
+    render(<Toolbar content={() => <div></div>} />);
+    expect(screen.getByTestId('hue-toolbar')).toBeVisible();
+  });
+  test('offers custom test-id', () => {
+    render(<Toolbar testId="custom-test-id" content={() => <div></div>} />);
+    expect(screen.getByTestId('custom-test-id')).toBeVisible();
+  });
+  test('can have custom classname', () => {
+    render(<Toolbar className="test-class" content={() => <div></div>} />);
+    expect(screen.getByRole('list')).toHaveClass('test-class');
+  });
+  test('has default height 40px', () => {
+    render(<Toolbar content={() => <div></div>} />);
+    expect(screen.getByRole('list')).toHaveStyle('--toolbar-height: 40px;');
+  });
+  test('can have a custom height', () => {
+    render(<Toolbar height="60px" content={() => <div></div>} />);
+    expect(screen.getByRole('list')).toHaveStyle('--toolbar-height: 60px;');
+  });
+
+  describe('ToolbarButton', () => {
+    test('renders a button wrapped in a list element to match the parent menu tag', () => {
+      render(<Toolbar content={() => <ToolbarButton />} />);
+      expect(screen.getByRole('button').closest('li')).toBeVisible();
+    });
+    test('renders a button name', () => {
+      render(<Toolbar content={() => <ToolbarButton> testbutton </ToolbarButton>} />);
+      expect(screen.getByRole('button', { name: 'testbutton' })).toBeVisible();
+    });
+    test('has default test-id "hue-toolbar-button" on wrapping li element', () => {
+      render(<Toolbar content={() => <ToolbarButton />} />);
+      expect(screen.getByTestId('hue-toolbar-button')).toBeVisible();
+    });
+    test('offers custom test-id on wrapping li element', () => {
+      render(<Toolbar content={() => <ToolbarButton testId="button-test-id" />} />);
+      expect(screen.getByTestId('button-test-id')).toBeVisible();
+    });
+    test('default test-id on button element', () => {
+      render(<Toolbar content={() => <ToolbarButton />} />);
+      expect(screen.getByTestId('hue-toolbar-button-btn')).toBeVisible();
+    });
+    test('offers custom test-id on button element', () => {
+      render(<Toolbar content={() => <ToolbarButton testId="button-test-id" />} />);
+      expect(screen.getByTestId('button-test-id-btn')).toBeVisible();
+    });
+
```

Comment:
Nit: inconsistent space between tests. I personally find it easier to read when there's a line between like here.

Reply:
fixed